apps/rush-lib/src/api/RushProjectConfiguration.ts

```diff
@@ -329,41 +328,60 @@ export class RushProjectConfiguration {
-                      `for project "${project.packageName}". The project output folder name "${outputFolderName}" in ` +
-                      `operation ${operationName} overlaps with other folder names in the same operation and with ` +
-                      `folder names in the following other operations: ${otherOperationNames.join(', ')}.`
-                  );
+      // For each phased command, check if any of its phases' output folders overlap.
+      for (const [, command] of repoCommandLineConfiguration.commands) {
+        if (command.commandKind === 'phased') {
+          const phasesOverlappingPathAnalyzer: OverlappingPathAnalyzer<string> =
+            new OverlappingPathAnalyzer<string>();
+
+          for (const operationName of command.phases) {
+            const operationSettings: IOperationSettings | undefined =
+              operationSettingsByOperationName.get(operationName);
+            if (operationSettings) {
+              if (operationSettings.outputFolderNames) {
+                for (const outputFolderName of operationSettings.outputFolderNames) {
+                  const overlappingOperationNames: string[] | undefined =
+                    phasesOverlappingPathAnalyzer.addPathAndGetFirstEncounteredLabels(
+                      outputFolderName,
+                      operationName
+                    );
+                  if (overlappingOperationNames) {
+                    const overlapsWithOwnOperation: boolean =
+                      overlappingOperationNames?.includes(operationName);
+                    if (overlapsWithOwnOperation) {
+                      if (overlappingOperationNames.length === 1) {
+                        terminal.writeErrorLine(
+                          `Invalid "${RUSH_PROJECT_CONFIGURATION_FILE.projectRelativeFilePath}" ` +
+                            `for project "${project.packageName}". The project output folder name "${outputFolderName}" in ` +
+                            `operation "${operationName}" overlaps with another folder name in the same operation.`
+                        );
+                      } else {
+                        const otherOperationNames: string[] = overlappingOperationNames.filter(
+                          (overlappingOperationName) => overlappingOperationName !== operationName
+                        );
+                        terminal.writeErrorLine(
+                          `Invalid "${RUSH_PROJECT_CONFIGURATION_FILE.projectRelativeFilePath}" ` +
+                            `for project "${project.packageName}". The project output folder name "${outputFolderName}" in ` +
+                            `operation "${operationName}" overlaps with other folder names in the same operation and with ` +
+                            'folder names in the following other operations invoked by the ' +
+                            `"${command.name}" command: ${otherOperationNames.join(', ')}.`
```

Comment:
```
Error: The "rush-project.json" config file defines two potentially simultaneous operations
whose "outputFolderNames" would overlap. Simultaneous operations should not delete
each other's output.  

Operations with overlapping output folders: "_phase:build", "_phase:lint"
Commands that invoke these operations: "build", "lint"
Potentially overlapping folders: "lib", "lib/public", "assets", "assets/public"
```

Reply:
Updated to

```
The "config/rush-project.json" config file in the "myproject" project defines two potentially
simultaneous operations whose "outputFolderNames" would overlap. Simultaneous operations should not
delete each other's output.

The "lib" path overlaps across these operations: "_phase:build", "_phase:lint"
```

Reply:
Minor suggestion:  "between" instead of "across"